lib/maxwell/adapter/util.ex

```diff
@@ -136,9 +136,24 @@ defmodule Maxwell.Adapter.Util do
   end
   defp multipart_body(:end), do: :eof
 
-  defp append_query_string(url, path, query)when query == %{}, do: url <> path
+  defp append_query_string(url, path, query)when query == %{}, do: url_joint_path(url, path)
   defp append_query_string(url, path, query) do
     query_string = URI.encode_query(query)
-    url <> path <> "?" <> query_string
+    url_path = url_joint_path(url ,path)
+    url_path <> "?" <> query_string
   end
+
+  defp url_joint_path(url, path) do
```

Comment:
Good idea.
But I was wrong about url and path
`put_path/1` is deprecated, use `new/1` or `new/2` followed by `put_path/2` instead.
Now:
url is parse by `URI.parse/1`, so url and path always ok if we don't use `put_path`,

https://github.com/zhongwencool/maxwell/blob/master/lib/maxwell/conn.ex#L127-L134

-----
So we just need update document in `maxwell.ex`